File: verify_ui.js
/**
 * Playwright UI 검증 스크립트
 * - 6개 UI 상태, 예측 결과, 단지 로드, 에러 처리 검증
 */
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage({ viewport: { width: 1440, height: 900 } });

  page.on('console', m => {
    if (m.type() === 'error') console.log('브라우저 콘솔 에러:', m.text());
  });

  await page.goto('http://localhost:3000');
  await page.waitForLoadState('networkidle');
  await new Promise(r => setTimeout(r, 1500));

  // 1. 초기 idle 상태
  const initState = await page.evaluate(
    () => document.getElementById('result-panel').dataset.uiState
  );
  console.log('1. 초기 UI 상태:', initState === 'idle' ? 'PASS (idle)' : 'FAIL: ' + initState);

  // 2. 지도 렌더링
  const mapCount = await page.locator('#map .leaflet-container').count();
  console.log('2. 지도 렌더링:', mapCount > 0 ? 'PASS' : 'FAIL (Leaflet 없음)');

  // 3. 구 선택 → 단지 로드
  await page.selectOption('#gu-select', '강남구');
  await new Promise(r => setTimeout(r, 2000));
  const cplxOpts = await page.evaluate(
    () => document.getElementById('complex-select').options.length
  );
  console.log('3. 단지 드롭다운 로드:', cplxOpts > 5 ? `PASS (${cplxOpts}개)` : 'FAIL: ' + cplxOpts);

  // 4. 예측 실행 → success 상태
  await page.fill('#area', '84');
  await page.fill('#floor', '10');
  await page.fill('#built', '2010');
  await page.click('#predict-btn');
  await new Promise(r => setTimeout(r, 2500));

  const afterState = await page.evaluate(
    () => document.getElementById('result-panel').dataset.uiState
  );
  console.log('4. 예측 후 UI 상태:', afterState === 'success' ? 'PASS (success)' : 'FAIL: ' + afterState);

  if (afterState === 'success') {
    const price  = await page.evaluate(() => document.getElementById('res-price-main').textContent);
    const sub    = await page.evaluate(() => document.getElementById('res-price-sub').textContent);
    const signal = await page.evaluate(() => document.getElementById('signal-badge').textContent);
    const conf   = await page.evaluate(() => document.getElementById('conf-pct-val').textContent);
    console.log('   예측가:', price);
    console.log('   예측 구간:', sub);
    console.log('   투자 신호:', signal);
    console.log('   신뢰도:', conf);
  }

  // 5. 피처 중요도 차트 렌더링
  const fiCanvas = await page.locator('#fi-chart').count();
  console.log('5. 피처 중요도 차트:', fiCanvas > 0 ? 'PASS' : 'FAIL');

  // 6. 추이 차트
  const trendsCanvas = await page.locator('#trends-chart').count();
  console.log('6. 추이 차트:', trendsCanvas > 0 ? 'PASS' : 'FAIL');

  // 7. 비교 패널
  const compareContent = await page.evaluate(
    () => document.getElementById('compare-content').style.display
  );
  console.log('7. 비교 패널:', compareContent === 'block' ? 'PASS (표시됨)' : 'FAIL: ' + compareContent);

  // 8. 스크린샷
  await page.screenshot({ path: '/tmp/apt_ui.png', fullPage: false });
  console.log('8. 스크린샷: /tmp/apt_ui.png');

  // 9. 에러 상태 (area=0)
  await page.fill('#area', '0');
  await page.click('#predict-btn');
  await new Promise(r => setTimeout(r, 600));
  const toast = await page.evaluate(() => document.getElementById('toast').textContent);
  const toastShow = await page.evaluate(() => document.getElementById('toast').className);
  console.log('9. 에러 토스트 (area=0):', toastShow.includes('show') ? `PASS ("${toast}")` : 'FAIL');

  // 10. 단지 선택 + 재예측
  await page.fill('#area', '84');
  await page.selectOption('#complex-select', { index: 1 }); // 1번째 단지 선택
  await page.click('#predict-btn');
  await new Promise(r => setTimeout(r, 2000));

  const state10 = await page.evaluate(
    () => document.getElementById('result-panel').dataset.uiState
  );
  const badge = await page.evaluate(() => document.getElementById('complex-badge-wrap').style.display);
  console.log('10. 단지 선택 예측:', state10 === 'success' ? 'PASS' : 'FAIL: ' + state10);
  console.log('    단지 배지 표시:', badge === 'block' ? 'PASS' : 'FAIL: ' + badge);

  // 스크린샷 (단지 선택 상태)
  await page.screenshot({ path: '/tmp/apt_ui_complex.png', fullPage: false });
  console.log('    단지 선택 스크린샷: /tmp/apt_ui_complex.png');

  await browser.close();
  console.log('\n전체 Playwright 검증 완료');
})().catch(e => {
  console.error('검증 실패:', e.message);
  process.exit(1);
});

File: verify_ui.py
"""Playwright UI 검증 — 6개 UI 상태 + 예측 흐름 + 단지 선택 완전 검증"""
from playwright.sync_api import sync_playwright
import time

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page(viewport={"width": 1440, "height": 900})
        errors = []
        page.on("console", lambda m: errors.append(m.text) if m.type == "error" else None)

        page.goto("http://localhost:3000")
        page.wait_for_load_state("networkidle", timeout=15000)
        time.sleep(1.5)

        s = page.evaluate("document.getElementById('result-panel').dataset.uiState")
        print(f"1. 초기 상태: {'PASS (idle)' if s=='idle' else 'FAIL: '+s}")

        cnt = page.locator(".leaflet-container").count()
        print(f"2. 지도: {'PASS' if cnt>0 else 'FAIL'}")

        page.select_option("#gu-select", "강남구")
        time.sleep(2.0)
        opts = page.evaluate("document.getElementById('complex-select').options.length")
        print(f"3. 단지 드롭다운: {'PASS ('+str(opts)+'개)' if opts>5 else 'FAIL: '+str(opts)}")

        page.fill("#area", "84")
        page.fill("#floor", "10")
        page.fill("#built", "2010")
        page.click("#predict-btn")
        time.sleep(2.5)

        s2 = page.evaluate("document.getElementById('result-panel').dataset.uiState")
        print(f"4. 예측 상태: {'PASS (success)' if s2=='success' else 'FAIL: '+s2}")

        if s2 == "success":
            price  = page.evaluate("document.getElementById('res-price-main').textContent")
            sub    = page.evaluate("document.getElementById('res-price-sub').textContent")
            signal = page.evaluate("document.getElementById('signal-badge').textContent")
            conf   = page.evaluate("document.getElementById('conf-pct-val').textContent")
            print(f"   예측가: {price}")
            print(f"   구간: {sub}")
            print(f"   신호: {signal}")
            print(f"   신뢰도: {conf}")

        fi_cnt = page.locator("#fi-chart").count()
        print(f"5. 피처 중요도 차트: {'PASS' if fi_cnt>0 else 'FAIL'}")

        tr_cnt = page.locator("#trends-chart").count()
        print(f"6. 추이 차트: {'PASS' if tr_cnt>0 else 'FAIL'}")

        cmp_vis = page.evaluate("document.getElementById('compare-content').style.display")
        print(f"7. 비교 패널: {'PASS' if cmp_vis=='block' else 'FAIL: '+cmp_vis}")

        page.screenshot(path="/tmp/apt_ui.png")
        print("8. 스크린샷 저장: /tmp/apt_ui.png")

        page.fill("#area", "0")
        page.click("#predict-btn")
        time.sleep(0.7)
        toast_cls = page.evaluate("document.getElementById('toast').className")
        toast_txt = page.evaluate("document.getElementById('toast').textContent")
        print(f"9. 에러 토스트: {'PASS' if 'show' in toast_cls else 'FAIL'} ({toast_txt!r})")

        page.fill("#area", "84")
        page.select_option("#complex-select", index=1)
        page.click("#predict-btn")
        time.sleep(2.5)
        s3 = page.evaluate("document.getElementById('result-panel').dataset.uiState")
        badge_d = page.evaluate("document.getElementById('complex-badge-wrap').style.display")
        badge_t = page.evaluate("document.getElementById('complex-badge-text').textContent")
        print(f"10. 단지 선택 예측: {'PASS' if s3=='success' else 'FAIL: '+s3}")
        print(f"    단지 배지: {'PASS ('+badge_t+')' if badge_d=='block' else 'FAIL: '+badge_d}")

        page.screenshot(path="/tmp/apt_ui_complex.png")
        print("    단지 선택 스크린샷: /tmp/apt_ui_complex.png")

        if errors:
            print(f"⚠ 브라우저 콘솔 에러 {len(errors)}건:", errors[:3])

        browser.close()
        print("\n전체 Playwright 검증 완료")

run()
